sycl/test/extensions/annotated_usm/annotated_usm.cpp

```diff
@@ -0,0 +1,192 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsyntax-only -Xclang -verify -Xclang -verify-ignore-unexpected=note %s
+// expected-no-diagnostics
+
+// Compile-time tests for annotated USM allocation functions
+
+#include <sycl/sycl.hpp>
+
+#include <complex>
+
+#include "fake_properties.hpp"
+
+// clang-format on
+
+using namespace sycl::ext::oneapi::experimental;
+using namespace sycl::ext::intel::experimental;
+using alloc = sycl::usm::alloc;
+
+namespace sycl {
+namespace ext::oneapi::experimental {
+
+// Make runtime property `foo` and `foz` valid for this test
+template <> struct IsRuntimePropertyValid<foo_key> : std::true_type {};
```

Comment:
This trait is used to determine if a runtime property is valid for the annotated malloc APIs, right?

If so, can you move it to the 'detail' namespace? It shouldn't be exposed to users.